src/api.rs

```diff
@@ -24,10 +24,10 @@ use tracing::{debug, error, info, trace};
 
 /// In the absence of a port supplied by the user via the config we will first try using this
 /// before using a random port.
-pub const DEFAULT_PORT_TO_TRY: u16 = 0;
+const DEFAULT_PORT_TO_TRY: u16 = 0;
```

Comment:
This should be 5483, I don't know why that changed, but 0 is any port (so is default). We try 5483 (live on tel keyboard) and the reason is all hell breaks loose then nodes will look for each other on known ports.

We can though do that later and remove this for now and just use the default (which is 0 -> any port) if no port givenSo I suggest get rid of this completely.

Reply:
I'll take this as a follow-up. I think it would be better to get rid of any cleverness about port binding from this crate. `qp2p` users should supply a port and deal with binding errors. `safe_network` can default to 5483 and fallback to 0 if necessary (though I'd still find that more surprising than it simply failing).

Reply:
I'll take this as a follow-up. I think it'd be best to remove port binding cleverness from this crate and have callers just set a port. We can then specify the default port (and fallback to random) in `safe_network`, which seems like the right place for it.

Reply:
> remove port binding cleverness from this crate and have callers just set a port
Agreed. The code actually does this btw. It tries the port specified by the caller, if nothing is specified it tries the port defined in this constant and if that fails it tries a random port (unless configured otherwise)